modules/src/ics07_tendermint/header.rs

```diff
@@ -11,26 +13,57 @@ use crate::ics02_client::header::AnyHeader;
 use crate::ics07_tendermint::error::{Error, Kind};
 use crate::ics24_host::identifier::ChainId;
 use crate::Height;
+use std::cmp::Ordering;
 
 /// Tendermint consensus header
-#[derive(Clone, Debug, PartialEq)] // TODO: Add Eq bound once present in tendermint-rs
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)] // TODO: Add Eq bound once present in tendermint-rs
 pub struct Header {
     pub signed_header: SignedHeader, // contains the commitment root
     pub validator_set: ValidatorSet, // the validator set that signed Header
     pub trusted_height: Height, // the height of a trusted header seen by client less than or equal to Header
     pub trusted_validator_set: ValidatorSet, // the last trusted validator set at trusted height
 }
 
+impl Header {
+    pub fn height(&self) -> Height {
+        Height::new(
+            ChainId::chain_version(self.signed_header.header.chain_id.as_str()),
+            u64::from(self.signed_header.header.height),
+        )
+    }
+    pub fn time(&self) -> Time {
+        self.signed_header.header.time
+    }
+
+    pub fn compatible_with(&self, ibc_client_header: &Header) -> bool {
+        let ibc_client_height = ibc_client_header.signed_header.header.height;
+        let self_header_height = self.signed_header.header.height;
+
+        match self_header_height.cmp(&&ibc_client_height) {
+            Ordering::Equal => {
+                // fork
+                self.signed_header.commit.block_id
+                    != ibc_client_header.signed_header.commit.block_id
+            }
+            Ordering::Greater => {
+                // BFT time violation
+                self.signed_header.header.time < ibc_client_header.signed_header.header.time
```

Comment:
Same here maybe ? The headers are compatible if the order on the heigh is the same as the order on the time. ```self.signed_header.header.time > ibc_client_header.signed_header.header.time```

Reply:
yes see ^